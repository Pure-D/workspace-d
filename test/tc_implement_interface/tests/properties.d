module properties;

class Foo : Bar
{
	private int m_tree;
	private int mCar;
	private int _heli;
	private int sharp;
	private int java;
}

interface Bar
{
	int tree() @property;
	void tree(int value) @property;
	int car() @property;
	void car(int foobar) @property;
	int heli() @property;
	void heli(int value) @property;
	int Sharp() @property;
	void Sharp(int value) @property;
	int getJava();
	void setJava(int value);

	int getter() @property;
	void setter(int value) @property;
}