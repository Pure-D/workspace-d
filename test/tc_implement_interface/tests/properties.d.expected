implement 33
int tree() @property
return m_tree
void tree(int value) @property
m_tree = value;

int car() @property
return mCar
void car(int foobar) @property
mCar = foobar;

int heli() @property
return _heli
void heli(int value) @property
_heli = value;

int Sharp() @property
return sharp
void Sharp(int value) @property
sharp = value;

int getJava()
return java
void setJava(int value)
java = value;

int getter() @property
void setter(int value) @property
